### ISSUE TITLE
在YACL上支持恶意Beaver三元组和shuffle算法

### ISSUE THREAD

**Issue by JosefinaYang** [secretflow/yacl#160]
In YACL, all existing algorithms are under the semi-honest model.

My question is: is it possible to implement a Beaver Triple and shuffling protocols in the malicious setting? We need those protocols as primitives to design a new protocol and use secretflow to test our protocol for benchmarking to publish academic papers.

The necessary knowledge that might be used in this issue includes cryptography, oblivious transfer, and secret shared shuffle.  
And let's consider the computational security parameter is $\kappa=128$.

Some related works are provided as follows.
[Ferret: Fast extension for correlated OT with small communication](https://dl.acm.org/doi/abs/10.1145/3372297.3417276). In Proceedings of the 2020 ACM SIGSAC Conference on Computer and Communications Security.
[Secret-shared shuffle](https://link.springer.com/chapter/10.1007/978-3-030-64840-4_12). ASIACRYPT 2020

**Comment by Jamie-Cui** [comment]
> implement ... Beaver Triple ... ?

Yes, the short answer is: it is possible to implement beaver's triple with Yacl's OT protocols in the malicious setting. We provide a malicious OTe (KOS with security fix), but i'm not currently having in mind the "paper" to actually implement triple with a malicious OT (you need some kind of consistency check).

> implement ... Secret-Shared Shuffle?

Yes, it only uses OT, prg and RO.

BTW, are you trying to implement your research work? We may provide additional help if you like.

**Comment by Jamie-Cui** [comment]
FYI, for security parameter, we are currently in the process of supporting explicit declaration of $kappa$ for different Yacl modules. And for your task, we can configure the low-level crypto to support at least 128-bit security.
